frontend/tests/controllers/UserGetTypesTest.php

```diff
@@ -0,0 +1,440 @@
+<?php
+/**
+ * Tests function getUserTypes
+ */
+class UserGetTypesTest extends \OmegaUp\Test\ControllerTestCase {
+    /**
+     * Get empty user types when user objectives are not yet established
+     */
+    public function testEmptyUserTypes() {
+        ['user' => $user, 'identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+        $login = self::login($identity);
+
+        $r = new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+        ]);
+
+        $userTypes = \OmegaUp\Controllers\User::getUserTypes(
+            $user->user_id,
+            $r
+        );
+        $this->assertEmpty($userTypes);
+    }
+
+    /**
+     * Test only admin and owner can get user types
+     */
+    public function testOnlyAllowedUsersCanGetUserTypes() {
+        ['identity' => $identity1] = \OmegaUp\Test\Factories\User::createUser();
+        ['identity' => $identity2] = \OmegaUp\Test\Factories\User::createUser();
+        $login = self::login($identity2);
+
+        $r = new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+        ]);
+
+        try {
+            \OmegaUp\Controllers\User::getUserTypes($identity1->user_id, $r);
+            $this->fail('Should have failed');
+        } catch (\OmegaUp\Exceptions\ForbiddenAccessException $e) {
+            $this->assertEquals('userNotAllowed', $e->getMessage());
+        }
+
+        //Admin can get user types
+        ['identity' => $identityAdmin] = \OmegaUp\Test\Factories\User::createAdminUser();
+        $adminLogin = self::login(($identityAdmin));
+        $r = new \OmegaUp\Request([
+            'auth_token' => $adminLogin->auth_token,
+        ]);
+
+        $userTypes = \OmegaUp\Controllers\User::getUserTypes(
+            $identity1->user_id,
+            $r
+        );
+        $this->assertEmpty($userTypes);
+    }
+
+    /**
+     * Get user types when learning objective is true
+     */
+    public function testLearningObjectiveUserTypes() {
+        ['user' => $user, 'identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+        $login = self::login($identity);
+
+        //should only get 'student' type
+        $r = new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+            'has_competitive_objective' => false,
+            'has_learning_objective' => true,
+            'has_scholar_objective' => true,
+            'has_teaching_objective' => false,
+        ]);
+        \OmegaUp\Controllers\User::apiUpdate($r);
```

Comment:
la convención es evitar declarar el request como variable independiente para evitar la tentación de pasarlo en varios requests:

```suggestion
        \OmegaUp\Controllers\User::apiUpdate(new \OmegaUp\Request([
            'auth_token' => $login->auth_token,
            'has_competitive_objective' => false,
            'has_learning_objective' => true,
            'has_scholar_objective' => true,
            'has_teaching_objective' => false,
        ]));
```